core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -878,15 +924,21 @@ class ReplicaManager(val config: KafkaConfig,
         else
           Set.empty[Partition]
 
+        leaderAndISRRequest.partitionStates.asScala.keys.foreach( topicPartition =>
```

Comment:
Never mind. This is still needed. If there is offline log dir, `getOrCreatePartition()` will create the partition object before `getOrCreateReplica()` fails with exception. In this case we need this code to remove the partition which doesn't have a valid local replica.